src/coreclr/jit/emitriscv64.cpp

```diff
@@ -1253,8 +1253,9 @@ void emitter::emitLoadImmediate(emitAttr size, regNumber reg, ssize_t imm)
     // Since ADDIW use sign extension fo immediate
     // we have to adjust higher 19 bit loaded by LUI
     // for case when low part is bigger than 0x800.
-    UINT32 high19 = (high31 + 0x800) >> 12;
+    INT32 high19 = ((int32_t)(high31 + 0x800)) >> 12;
 
+    assert(isValidSimm20(high19));
```

Comment:
NIT: This assert is redundant since the immediate value will also be checked in the `emitIns_R_I` function

Reply:
Thank you! Removed.